src/common/structs/is-custom-type.ts

```diff
@@ -0,0 +1,10 @@
+import { refine, string } from "superstruct";
+
+const isCustom = (value: string) => {
+  if (value.startsWith("custom:")) {
+    return true;
+  }
+  return false;
+};
+
+export const custom = () => refine(string(), "custom element type", isCustom);
```

Comment:
We should name functions as if they all are stored in a single file. That helps most with auto complete and understanding when reading code